internal/stacktrace/stacktrace.go

```diff
@@ -0,0 +1,195 @@
+		}
+	}
+
+	defaultTopFrameDepth = defaultMaxDepth / 4
+
+	if env := os.Getenv(stackTraceDisabledEnvVar); env != "" {
+		if e, err := strconv.ParseBool(env); err == nil {
+			enabled = e
+		}
+	}
+}
+
+// StackTrace is intended to be sent over the span tag `_dd.stack`, the first frame is the current frame
+type StackTrace []StackFrame
+
+// StackFrame represents a single frame in the stack trace
+type StackFrame struct {
+	Index     uint32 `msg:"id"`                   // Index of the frame (0 = top of the stack)
+	Text      string `msg:"text,omitempty"`       // Text version of the stackframe as a string
+	File      string `msg:"file,omitempty"`       // File name where the code line is
+	Line      uint32 `msg:"line,omitempty"`       // Line number in the context of the file where the code is
+	Column    uint32 `msg:"column,omitempty"`     // Column where the code ran is
+	Namespace string `msg:"namespace,omitempty"`  // Namespace is the fully qualified name of the package where the code is
+	ClassName string `msg:"class_name,omitempty"` // ClassName is the fully qualified name of the class where the line of code is
+	Function  string `msg:"function,omitempty"`   // Function is the fully qualified name of the function where the line of code is
+}
+
+// getPackageFromSymbol returns the package name from a function name
+// ex: gopkg.in/DataDog/dd-trace-go.v1/internal/stacktrace.(*Event).NewException -> gopkg.in/DataDog/dd-trace-go.v1/internal/stacktrace
+func getPackageFromSymbol(name string) string {
+	// Find the last slash
+	index := strings.LastIndex(name, "/")
+	if index == -1 {
+		index = 0
+	}
+
+	// Find the first dot after the last slash
+	index += strings.Index(name[index:], ".")
+	return name[:index]
+}
+
+// getMethodReceiverFromSymbol returns the receiver of a method from a function name
+// ex: gopkg.in/DataDog/dd-trace-go.v1/internal/stacktrace.(*Event).NewException -> *Event
+func getMethodReceiverFromSymbol(name string) string {
+	startIndex := strings.Index(name, "(")
+	if startIndex == -1 {
+		return ""
+	}
+
+	endIndex := strings.LastIndex(name, ")")
+	if endIndex == -1 || endIndex <= startIndex+1 {
+		return ""
+	}
+
+	return name[startIndex+1 : endIndex]
+}
+
+// getFunctionNameFromSymbol returns the function name from a symbol
+// ex: gopkg.in/DataDog/dd-trace-go.v1/internal/stacktrace.(*Event).NewException -> NewException
+func getFunctionNameFromSymbol(name string) string {
+	index := strings.LastIndex(name, ")")
+
+	// If it's a method, we just have to get what's after the last closing parenthesis
+	if index != -1 && index+2 < len(name) {
+		return name[index+2:]
+	}
+
+	// If it's a function, we have to get past the package name and get the end of the string
+	index = strings.LastIndex(name, "/")
+	if index == -1 {
+		index = 0
+	}
+
+	index += strings.Index(name[index:], ".")
+	return name[index+1:]
+}
+
+// Take Create a new stack trace from the current call stack
+func Take() StackTrace {
+	return TakeWithSkip(4)
```

Comment:
This `4` here looks odd... Maybe this should be documented? I suspect this bakes in a bunch of assumptions about the call depth of this function, and those are likely easy to break...

Reply:
Forgot to use `defaultCallerSkip`

Reply:
But yes this whole thing is easy to break. Well, "break" here means take ~1 more or less frames than necessary into account.

But yes, we have to think of something to skip unwanted frames. @Julio-Guerra had some ideas but I don't thing they should be integrated directly as-is in the stacktrace package who should have only the job of take the stacktrace to the user. We could definitively buff it up with some regex filtering or other generic stuff like this.